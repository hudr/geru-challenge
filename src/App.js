import React, { Component, Fragment } from 'react'
import PropTypes from 'prop-types'

import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import * as TodoActions from './redux/actions/todo'

import Header from './components/Header'
import Modal from './components/Modal'
import EmptyCards from './components/EmptyCards'
import TodoCards from './components/TodoCards'

class App extends Component {
  async componentDidMount() {
    this.props.getTodos()
  }

  render() {
    const { todos, isModalOpen } = this.props

    return (
      <Fragment>
        <Header />
        {isModalOpen && <Modal />}
        {todos.length === 0 ? <EmptyCards /> : <TodoCards />}
      </Fragment>
    )
  }
}

const mapStateToProps = state => ({
  todos: state.todo.todos,
  isModalOpen: state.todo.isModalOpen
})

const mapDispatchToProps = dispatch => bindActionCreators(TodoActions, dispatch)

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App)

App.propTypes = {
  getTodos: PropTypes.func,
  todos: PropTypes.array.isRequired,
  isModalOpen: PropTypes.bool.isRequired
}

export { App }
