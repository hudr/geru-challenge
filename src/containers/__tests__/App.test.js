import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { App } from '../../App'

const props = {
  todos: [],
  getTodos: jest.fn(),
  isModalOpen: false
}

describe('<App />', () => {
  it('renders without crashing given props', () => {
    const wrapper = shallow(<App {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders with modal and todocards', () => {
    props.isModalOpen = true
    props.todos = ['Title', 'Description']
    const wrapper = shallow(<App {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
