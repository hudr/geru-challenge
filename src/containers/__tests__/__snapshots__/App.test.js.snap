// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders with modal and todocards 1`] = `
<Fragment>
  <Connect(Header) />
  <Connect(WithWidth(WithMobileDialog)) />
  <Connect(TodoCards) />
</Fragment>
`;

exports[`<App /> renders without crashing given props 1`] = `
<Fragment>
  <Connect(Header) />
  <EmptyCards />
</Fragment>
`;
